engine/src/main/java/com/redhat/cloud/notifications/processors/email/aggregators/AdvisorEmailAggregator.java

```diff
@@ -0,0 +1,197 @@
+            "total_risk": 3
+        }
+    }
+}
+ */
+
+public class AdvisorEmailAggregator extends AbstractEmailPayloadAggregator {
+
+    // Advisor event types
+    public static final String NEW_RECOMMENDATION = "new-recommendation";
+    public static final String RESOLVED_RECOMMENDATION = "resolved-recommendation";
+    public static final String DEACTIVATED_RECOMMENDATION = "deactivated-recommendation";
+
+    // Advisor events aggregator data contents
+    public static final String ADVISOR_KEY = "advisor";
+    public static final String NEW_RECOMMENDATIONS = "new_recommendations";
+    public static final String RESOLVED_RECOMMENDATIONS = "resolved_recommendations";
+    public static final String DEACTIVATED_RECOMMENDATIONS = "deactivated_recommendations";
+    public static final String RULE_ID = "rule_id";
+    public static final String RULE_DESCRIPTION = "rule_description";
+    public static final String TOTAL_RISK = "total_risk";
+    public static final String HAS_INCIDENT = "has_incident";
+    public static final String RULE_URL = "rule_url";
+    public static final String CONTENT_SYSTEM_COUNT = "systems";
+
+    // Notification common
+    private static final String EVENT_TYPE_KEY = "event_type";
+    private static final String EVENTS_KEY = "events";
+    private static final String PAYLOAD_KEY = "payload";
+
+    private static final Set<String> EVENT_TYPES = new HashSet<>(Arrays.asList(
+            NEW_RECOMMENDATION, RESOLVED_RECOMMENDATION,
+            DEACTIVATED_RECOMMENDATION
+    ));
+
+    // Advisor final payload helpers
+    private final Map<String, Map<String, Object>> newRecommendations = new HashMap<>();
+    private final Map<String, Map<String, Object>> resolvedRecommendations = new HashMap<>();
+    private final Map<String, Map<String, String>> deactivatedRecommendations = new HashMap<>();
+
+    public AdvisorEmailAggregator() {
+        JsonObject advisor = new JsonObject();
+        advisor.put(NEW_RECOMMENDATIONS, newRecommendations);
+        advisor.put(RESOLVED_RECOMMENDATIONS, resolvedRecommendations);
+        advisor.put(DEACTIVATED_RECOMMENDATIONS, deactivatedRecommendations);
+
+        context.put(ADVISOR_KEY, advisor);
+    }
+
+    @Override
+    void processEmailAggregation(EmailAggregation notification) {
+        JsonObject notifPayload = notification.getPayload();
+        String eventType = notifPayload.getString(EVENT_TYPE_KEY);
+        /* Major assumption: that there is at most one event per day from each
+         * rule,system pair - i.e. that the tuple (day, rule, system) is
+         * unique for new, and resolved, events.  This is generally true,
+         * because a rule would have to be new twice or resolved twice - i.e.
+         * new, then resolved, then occur again - on the same system in order
+         * for this to not be wrong.  This means we don't need to remember the
+         * inventory ID of each system impacted by or resolving a rule. */
```

Comment:
@PaulWay If you change your mind about this, please let me know and I'll help you change the aggregation logic.